File: software/examples/javascript/ExampleThreshold.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletMoisture = require('Tinkerforge/BrickletMoisture');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'iB4';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var m = new BrickletMoisture(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Set Period for moisture callback to 1s (1000ms)
        // Note: The moisture callback is only called every second if the 
        // moisture has changed since the last call!
        m.setMoistureCallbackPeriod(1000);
        // Get threshold callbacks with a debounce time of 1 seconds (1000ms)
        m.setDebouncePeriod(1000);
        // Configure threshold for "greater than 200"
        m.setMoistureCallbackThreshold('>', 200, 0);      
    }
);

// Register threshold reached callback
m.on(BrickletMoisture.CALLBACK_MOISTURE_REACHED,
    // Callback for moisture value greater than 200
    function(moisture) {
        console.log('Moisture Value: '+moisture);
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/c/example_threshold.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_moisture.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

// Callback function for moisture value greater than 200
void cb_moisture_reached(uint16_t moisture, void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("Moisture Value: %d\n", moisture);
}

int main() {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	Moisture m;
	moisture_create(&m, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		exit(1);
	}
	// Don't use device before ipcon is connected

	// Get threshold callbacks with a debounce time of 1 second (1000ms)
	moisture_set_debounce_period(&m, 1000);

	// Register threshold reached callback to function cb_moisture_reached
	moisture_register_callback(&m,
	                           MOISTURE_CALLBACK_MOISTURE_REACHED,
	                           (void *)cb_moisture_reached,
	                           NULL);

	// Configure threshold for "greater than 200"
	moisture_set_moisture_callback_threshold(&m, '>', 200, 0);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
}
